pkg/api/helper/imageextraction.go

```diff
@@ -0,0 +1,75 @@
+package helper
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/load"
+)
+
+// GetAllImageStreamTags returns a deduplicated and unsorted list of all ImageStreamTags
+// referenced anywhere in a config. It only returns their namespace and name and drops the
+// cluster field, as we plan to remove that.
+func GetAllImageStreamTags(config load.ByOrgRepo) ([]types.NamespacedName, error) {
+	result := sets.String{}
+	for _, org := range config {
+		for _, repo := range org {
+			for _, cfg := range repo {
+				imageStreamTagReferenceMapIntoSet(cfg.BaseImages, &result)
+				imageStreamTagReferenceMapIntoSet(cfg.BaseRPMImages, &result)
+				if cfg.BuildRootImage != nil && cfg.BuildRootImage.ImageStreamTagReference != nil {
+					result.Insert(imageStreamTagReferenceToString(*cfg.BuildRootImage.ImageStreamTagReference))
+				}
+
+				for _, rawStep := range cfg.RawSteps {
+					if rawStep.InputImageTagStepConfiguration != nil {
+						result.Insert(imageStreamTagReferenceToString(rawStep.InputImageTagStepConfiguration.BaseImage))
+					}
+					if rawStep.SourceStepConfiguration != nil {
+						result.Insert(imageStreamTagReferenceToString(rawStep.SourceStepConfiguration.ClonerefsImage))
+					}
+					// TODO: Is this correct?
+					if rawStep.OutputImageTagStepConfiguration != nil {
```

Comment:
@bbguimaraes do you know?

Reply:
These are images that will be built from the configuration, so they should not be included if this PR is part of https://issues.redhat.com/browse/DPTP-1062.

We should rename the method to make it clear that only input ISTs will be included.